File: py/1435.py
while True:
    n = int(input())

    if n == 0:
        break

    length = n // 2
    max_count = 1
    odd = False

    if n % 2 != 0:
        odd = True
        length += 1

    matrix = [[1 for i in range(n)] for i in range(length)]

    for i in range(length):
        count = 1

        for j in range(n // 2):
            matrix[i][j] = count

            if count != max_count:
                count += 1

        count = 1

        for j in range(n - 1, n // 2 - 1, -1):
            matrix[i][j] = count

            if count != max_count:
                count += 1

        max_count += 1

    if odd:
        matrix[n // 2][n // 2] = length

    for i in matrix:
        r = ' '
        print(r.join([f'{j:3}' for j in i]))

    for i in range(n // 2 - 1, -1, -1):
        r = ' '
        print(r.join([f'{j:3}' for j in matrix[i]]))

    print('')

File: cpp/1435.cpp
#include <iostream>
#include <stdio.h>

using namespace std;

int main()
{
    int n;

    while (true) {
        cin >> n;
        if (n == 0) break;

        int length = n / 2, count = 1, max_count = 1, i, j;
        bool odd = false;

        if (n % 2 != 0) {
            odd = true;
            length++;
        }

        int matrix[length][n];

        for (i = 0; i < length; i++) {
            for (j = 0; j < n; j++) {
                matrix[i][j] = 1;
            }
        }

        for (i = 0; i < length; i++) {
            count = 1;

            for (j = 0; j < n / 2; j++) {
                matrix[i][j] = count;
                if (count != max_count) count++;
            }

            count = 1;

            for (j = n - 1; j > n / 2 - 1; j--) {
                matrix[i][j] = count;
                if (count != max_count) count++;
            }

            max_count++;
        }

        if (odd) matrix[n / 2][n / 2] = length;

        for (i = 0; i < length; i++) {
            for (j = 0; j < n - 1; j++) {
                printf("%3d ", matrix[i][j]);
            }

            printf("%3d\n", matrix[i][j]);
        }

        for (i = n / 2 - 1; i > -1; i--) {
            for (j = 0; j < n - 1; j++) {
                printf("%3d ", matrix[i][j]);
            }

            printf("%3d\n", matrix[i][j]);
        }

        cout << '\n';
    }

    return 0;
}